types/knockout.mapping/index.d.ts

```diff
@@ -4,7 +4,7 @@
 //                 Mathias Lorenzen <https://github.com/ffMathy>
 //                 Leonardo Lombardi <https://github.com/ltlombardi>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 2.3
+// TypeScript Version: 2.8
```

Comment:
I would suggest to have redirection of types and leave old version for compatibility.
As I know most users of ko is considering it legacy code with older ts.
So we should be careful with uplifting ts requirements.

Reply:
I don't know what you mean by: "redirection of types".

I thought long and hard about upping the TS version, but the only way for this typing to work is with version 2.8. 

It's my understanding that when someone needs and older version of a typing, he needs to check the git history and get the version that satisfy he's needs.

There is also a way to command to npm to download a specific version of the typings correlated with the TS version using `npm install @types/knockout.mapping@ts2.2`
Although I know this command is not popular.

[Examples of versions from the  "versions" in npm site:](https://www.npmjs.com/package/@types/knockout.mapping). 
![image](https://user-images.githubusercontent.com/898124/60127481-af004680-9766-11e9-8875-305ca603ea6c.png)



I don't think it's a good idea to limit development and usage of new features, because the userbase is using older versions. :disappointed:

Reply:
I'm sorry, I should've been clearer. I mean this types redirection: https://github.com/microsoft/TypeScript/wiki/What%27s-new-in-TypeScript#version-selection-with-typesversions

So it allows to have types for lower versions of TS as well as newer versions, in the same package.

Reply:
The problem I see is that it only works in TS 3.1 or above. So people with TS 2.8 thru 3.0 would be getting the older knockout.mapping file. Basically the version present in the PR will only be available to TS 3.1 users or above. 

I guess this is acceptable, but I really believe this is not the way to do it.

My reasoning:
1) If the project doesn't use a package delivery system, the dev already included the `.d.ts` file in it's project repository, so this PR has no harm for the legacy project and it's developer.
2) If the project uses Nuget, again, this changes won't affect him. He is stuck to the typings version in his projects.
3) If the project uses NPM, with a specific version set, again, this changes won't affect him. He is stuck to the typings version in his projects.
4) If the project uses NPM, without any version established, then this is the devs fault / decision to use the latest version and deal with the consequences. 

So my conclusion is that this won't really harm anyone already using knockout.mapping. 
Someone that will start to use now, will be actively developing on a legacy code... so It won't really be legacy, will it? That person would be more than better off upping the TSC version. But again he is not required, he can just set a specific version in NPM packages.json and be happy.

Reply:
Hey @Igorbek , the bot wrongly things there has been no activity on this PR and will close it tomorrow. Can you put a hold on it?

Reply:
@ltlombardi I think it should be quite easy to follow suggesting described here: https://github.com/DefinitelyTyped/DefinitelyTyped#i-want-to-use-features-from-typescript-31-or-above

I still think that there're projects that use older versions of TS and wouldn't upgrade.
However, I don't think I can prevent from upgrading to new TS now, and if any update for older version is required in the future, we can make a new redirection folder.